### PR TITLE
Add support for gzip compression

### PR DESCRIPTION
**Pull request by ODudek**
*Issue #, if available:*

*Description of changes:*

Send data to kinesis using gzip compression

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,8 @@
 # CHANGELOG
 
+## 3.4.2
```
Comment (simukappu) [inline CHANGELOG.md:3]: We should handle adding feature and version upgrade separately. Can you remove this version upgrade to 3.4.2 from your PR?

File: lib/fluent_plugin_kinesis/version.rb [hunk 2]
```diff
@@ -13,5 +13,5 @@
 # language governing permissions and limitations under the License.
 
 module FluentPluginKinesis
-  VERSION = '3.4.1'
+  VERSION = '3.4.2'
```
Comment (simukappu) [inline lib/fluent_plugin_kinesis/version.rb:16]: We should handle adding feature and version upgrade separately. Can you remove this version upgrade to 3.4.2 from your PR?

**Review by simukappu — CHANGES_REQUESTED** [review]
We should handle adding feature and version upgrade separately. Can you remove version upgrade to 3.4.2 from your PR? The gzip compression feature you added looks good. Thank you for your contribution!

File: lib/fluent_plugin_kinesis/version.rb (hunk 2 above)
Comment (ODudek) [reply lib/fluent_plugin_kinesis/version.rb]: Sure, changes reverted

**Review by simukappu — APPROVED** [review]
LGTM

**Pull request merged**